src/test/java/com/wikia/webdriver/testcases/adstests/TestFloorAdhesion.java

```diff
@@ -0,0 +1,90 @@
+import com.wikia.webdriver.common.templates.TemplateDontLogout;
+import com.wikia.webdriver.pageobjectsfactory.pageobject.adsbase.AdsBaseObject;
+import com.wikia.webdriver.pageobjectsfactory.pageobject.adsbase.mobile.MobileAdsBaseObject;
+
+import org.openqa.selenium.By;
+import org.openqa.selenium.WebElement;
+import org.testng.annotations.Test;
+
+public class TestFloorAdhesion extends TemplateDontLogout {
+    private final String FLOOR_ADHESION_SELECTOR = "#ext-wikia-adEngine-template-footer";
+    private final String FLOOR_ADHESION_AD_SELECTOR = "#ext-wikia-adEngine-template-footer .ad";
+    private final String
+            FLOOR_ADHESION_CLOSE_SELECTOR = "#ext-wikia-adEngine-template-footer .close";
+    private final String WIKIA_BAR_SELECTOR = "#WikiaBar";
+
+    @Test(
+            dataProvider = "floorAdhesionOnOasis",
+            dataProviderClass = AdsDataProvider.class,
+            groups = {"TestFloorAdhesionOasis"}
+    )
+    public void testFloorAdhesionOnOasis(
+            String wikiName,
+            String article,
+            String floorAdhesionModalSelector,
+            String floorAdhesionModalCloseSelector
+    ) {
+        String testPage = urlBuilder.getUrlForPath(wikiName, article);
+        AdsBaseObject wikiPage = new AdsBaseObject(driver, testPage);
+
+        verifyFloorAdhesionIsVisible(wikiPage);
+
+        // verify there is no Wikia Bar
+        WebElement wikiaBar = driver.findElement(By.cssSelector(WIKIA_BAR_SELECTOR));
+        wikiPage.waitForElementNotVisibleByElement(wikiaBar);
+        PageObjectLogging.log(
+                "testFloorAdhesionOnOasis",
+                "WikiaBar not visible",
+                true
+        );
+
+        verifyClickingUnitOpensLightbox(wikiPage, floorAdhesionModalSelector);
+        verifyClosingWorks(floorAdhesionModalCloseSelector);
+    }
+
+    @Test(
+            dataProvider = "floorAdhesionOnMercury",
+            dataProviderClass = MobileAdsDataProvider.class,
+            groups = {"TestFloorAdhesionMercury", "MercuryAds"}
+    )
+    public void floorAdhesionOnMercury(
+            String wikiName,
+            String article,
+            String floorAdhesionModalSelector,
+            String floorAdhesionModalCloseSelector
+    ) {
+        String testPage = urlBuilder.getUrlForPath(wikiName, article);
+        MobileAdsBaseObject wikiPage = new MobileAdsBaseObject(driver, testPage);
+
+        verifyFloorAdhesionIsVisible(wikiPage);
+        verifyClickingUnitOpensLightbox(wikiPage, floorAdhesionModalSelector);
+        verifyClosingWorks(floorAdhesionModalCloseSelector);
+    }
+
+    private void verifyFloorAdhesionIsVisible(AdsBaseObject wikiPage) {
+        wikiPage.waitForElementByCss(FLOOR_ADHESION_SELECTOR);
+        PageObjectLogging.log(
+                "testFloorAdhesionOnOasis",
+                "Floor Adhesion unit visible",
+                true
+        );
+    }
+
+    private void verifyClickingUnitOpensLightbox(
```

Comment:
Thanks for the hint with `PageObjectLogging.log()` I'll implement it right away! :) 

I'm not sure about having the selectors and those methods in other class. @PiotrGackowski suggested `AdsBaseObject` or `MobileAdsBaseObject`. I'm not a big fan of this solution unless other test classes use them - right now there are no other test classes which need either the selectors nor the methods. What are the reasons to move the stuff somewhere else?